packages/linter/src/generators/lint-project/lint-project.ts

```diff
@@ -177,6 +195,21 @@ function isBuildableLibraryProject(
   );
 }
 
+function isJsAnalyzeSourceFilesEnabled(tree: Tree): boolean {
+  const nxJson = readJson<NxJsonConfiguration>(tree, 'nx.json');
+  const jsPluginConfig = nxJson.pluginsConfig?.['@nx/js'] as {
+    analyzeSourceFiles?: boolean;
+  };
+
+  if (nxJson.extends === 'nx/presets/npm.json') {
+    // npm preset turns off analyzeSourceFiles, but check if user has overridden it.
+    return jsPluginConfig?.analyzeSourceFiles ?? false;
+  }
+
+  // By default the JS plugin enables analyzeSourceFiles as long as user did not override it.
+  return jsPluginConfig?.analyzeSourceFiles ?? true;
+}
```

Comment:
This can be shortened to
```suggestion
  return jsPluginConfig?.analyzeSourceFiles ?? nxJson.extends !== 'nx/presets/npm.json'
```